solidity/test/bridge/Bridge.Frauds.test.ts

```diff
@@ -0,0 +1,1528 @@
+            state: walletState.Live,
+          })
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            bridge
+              .connect(thirdParty)
+              .submitFraudChallenge(
+                walletPublicKey,
+                data.sighash,
+                data.signature,
+                {
+                  value: (await bridge.fraudChallengeDepositAmount()).sub(1),
+                }
+              )
+          ).to.be.revertedWith("The amount of ETH deposited is too low")
+        })
+      })
+    })
+
+    context("when the wallet is in MovingFunds state", () => {
+      before(async () => {
+        await createSnapshot()
+        await bridge.setWallet(walletPublicKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          state: walletState.Live,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should succeed", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              walletPublicKey,
+              data.sighash,
+              data.signature,
+              {
+                value: await bridge.fraudChallengeDepositAmount(),
+              }
+            )
+        ).to.not.be.reverted
+      })
+    })
+
+    context("when the wallet is in neither Live nor MovingFunds state", () => {
+      before(async () => {
+        await createSnapshot()
+        await bridge.setWallet(walletPublicKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          state: walletState.Closed,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              walletPublicKey,
+              data.sighash,
+              data.signature,
+              {
+                value: await bridge.fraudChallengeDepositAmount(),
+              }
+            )
+        ).to.be.revertedWith("Wallet is neither in Live nor MovingFunds state")
+      })
+    })
+  })
+
+  describe("defeatFraudChallenge", () => {
+    context("when the challenge exists", () => {
+      context("when the challenge is open", () => {
+        context("when the sighash type is correct", () => {
+          context("when the input is non-witness", () => {
+            context("when the transaction has single input", () => {
+              context(
+                "when the inputs are marked correctly spent in the Bridge",
```

Comment:
The challenge is always per single input so maybe `when the inputs are marked` -> `when the input is marked`? Same for all other scenarios below.